### PR TITLE
Banner code cleanups

### PR DESCRIPTION
**Pull request by Broxzier**
While working on #7580 I noticed that the banner's `colour` field is used to store ride indices. I wasn't sure if this was a bug, and upon further inspection I saw it was correct, just confusing. I've changes the `colour` field to be a union tied with `ride_index`.

While working on this, I also made some of the ride_index getters return uint8 instead of sint32, and did the same for banners later. For banners, there also was a small issue where the 250th banner would be spawned, but not initialized properly, as the code checked for `MAX_BANNER` instead of `BANNER_INDEX_NULL`.

### CODE REVIEW

File: src/openrct2/world/Banner.h [hunk 1]
```diff
@@ -22,22 +22,28 @@
 
 #define BANNER_NULL 255
 #define MAX_BANNERS 250
-#define BANNER_INDEX_NULL -1
+#define BANNER_INDEX_NULL ((uint8)-1)
```
Comment (Gymnasiast) [inline src/openrct2/world/Banner.h]: What's the difference between `BANNER_NULL` and `BANNER_INDEX_NULL`?

Comment (Broxzier) [reply src/openrct2/world/Banner.h]: `BANNER_NULL` is used for slots in `rct_banner gBanners[]` that are not in use.
`BANNER_INDEX_NULL` is much like `SPRITE_INDEX_NULL`, used by tile elements and sprites to refer to an entry in the banner list.

File: src/openrct2/world/Banner.cpp [hunk 2]
```diff
@@ -44,21 +44,17 @@ rct_banner gBanners[MAX_BANNERS];
  *
  *  rct2: 0x006B7EAB
  */
-static sint32 banner_get_ride_index_at(sint32 x, sint32 y, sint32 z)
+static uint8 banner_get_ride_index_at(sint32 x, sint32 y, sint32 z)
```
Comment (Gymnasiast) [inline src/openrct2/world/Banner.cpp]: Consider that we will expand the amount of rides possible in the near future, is returning the ride index as an uint8 really a good idea?

Comment (AaronVanGeffen) [reply src/openrct2/world/Banner.cpp]: I'd say that, until we do, it's the valid thing to do.

Comment (Broxzier) [reply src/openrct2/world/Banner.cpp]: If expanded in the future, it _should_ generate a warning, though now I think about it I do believe these warnings are disabled. It'd probably be nicest to have `ride_index` and `banner_index` types.

Comment (Broxzier) [reply src/openrct2/world/Banner.cpp]: I've added the new `BannerIndex` type and use it in all the places I could find, but left out `RideIndex` for now, as it's out of scope for this PR.

File: src/openrct2/world/Banner.h [hunk 3]
```diff
@@ -22,22 +22,28 @@
 
 #define BANNER_NULL 255
 #define MAX_BANNERS 250
-#define BANNER_INDEX_NULL -1
+#define BANNER_INDEX_NULL ((BannerIndex)-1)
```
Comment (Gymnasiast) [inline src/openrct2/world/Banner.h]: Perhaps a good moment to use a C++ constexpr here?

Comment (Broxzier) [reply src/openrct2/world/Banner.h]: To my surprise, `MAX_BANNERS` can be made constexpr too, even while it's used for array sizes! I'm sure I bumped into a problem with these before, but that must have been an older C++ version then.

File: src/openrct2/world/Banner.cpp [hunk 4]
```diff
@@ -441,18 +436,19 @@ sint32 banner_get_closest_ride_index(sint32 x, sint32 y, sint32 z)
...
-        if (rideIndex != -1)
+        uint8 rideIndex = banner_get_ride_index_at(x + NeighbourCheckOrder[i].x, y + NeighbourCheckOrder[i].y, z);
+        if (rideIndex != RIDE_ID_NULL)
         {
             return rideIndex;
         }
     }
 
-    rideIndex = -1;
+    sint32 index;
```
Comment (HaasJona) [inline src/openrct2/world/Banner.cpp]: I believe you can make index an `uint8` here, then you can eliminate the case later. Otherwise it looks good to me.

**Pull request merged**